realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -605,7 +625,9 @@ public void onResult(int count) {
                 RealmMigration realmMigration = (migration == null) ? configuration.getMigration() : migration;
                 DynamicRealm realm = null;
                 try {
-                    realm = DynamicRealm.getInstance(configuration);
+                    // Create a DynamicRealm WITHOUT putting it into a RealmCache to avoid recursive locks and call init
+                    // steps multiple times (copy asset file / initialData transaction).
+                    realm = DynamicRealm.createInstance(configuration);
```

Comment:
Wouldn't this theoretically make it possible to open a Realm inside a migration block with a different configuration for the same path? If there are clear benefits of doing it this way and it makes the code much cleaner, not catching that particular error is probably fine.